### PR TITLE
Resolved a significant performance degradation in `BeEquivalentTo`

### PR DESCRIPTION
**Pull request by dennisdoomen**
This was caused by the increased dependency on the CallerIdentifier, and the additional work caller identification takes in v6.

#1657

### CODE REVIEW

File: Tests/FluentAssertions.Specs/Equivalency/CollectionEquivalencySpecs.cs [hunk 1]
```diff
@@ -2830,5 +2830,77 @@ public static IEnumerable<object[]> ArrayTestData()
                    from y in arrays
                    select new object[] { x, y };
         }
+
+        [Fact]
+        public void Comparing_lots_of_complex_objects_should_still_be_fast()
```
Comment (jnyrup) [inline Tests/FluentAssertions.Specs/Equivalency/CollectionEquivalencySpecs.cs:2835]: Failed in pipeline

**Review by jnyrup — APPROVED** [review]
Benchmark results:

**master:**
```
|              Runtime |  N |            Mean |         Error |        StdDev |       Gen 0 |     Gen 1 |    Allocated |
|--------------------- |--- |----------------:|--------------:|--------------:|------------:|----------:|-------------:|
|             .NET 5.0 |  1 |        57.09 us |      0.299 us |      0.279 us |     11.1694 |    0.3052 |        69 KB |
| .NET Framework 4.7.2 |  1 |     2,025.39 us |     10.364 us |      9.695 us |    136.7188 |    7.8125 |       855 KB |
|             .NET 5.0 | 10 |   165,775.29 us |  1,407.268 us |  1,316.360 us |  11000.0000 | 1000.0000 |    68,264 KB |
| .NET Framework 4.7.2 | 10 |   216,483.12 us |    687.123 us |    573.779 us |  14333.3333 | 1000.0000 |    88,899 KB |
|             .NET 5.0 | 50 | 4,604,075.87 us | 28,517.080 us | 26,674.895 us | 308000.0000 | 1000.0000 | 1,892,259 KB |
| .NET Framework 4.7.2 | 50 | 5,552,894.02 us | 47,078.673 us | 41,734.018 us | 365000.0000 | 1000.0000 | 2,243,784 KB |
```

**PR 1660**
```
|              Runtime |  N |          Mean |        Error |       StdDev |      Gen 0 |     Gen 1 |  Allocated |
|--------------------- |--- |--------------:|-------------:|-------------:|-----------:|----------:|-----------:|
|             .NET 5.0 |  1 |      56.26 us |     0.351 us |     0.328 us |    11.1694 |    0.3052 |      69 KB |
| .NET Framework 4.7.2 |  1 |     277.59 us |     3.679 us |     3.441 us |    28.8086 |    0.9766 |     178 KB |
|             .NET 5.0 | 10 |  11,297.99 us |    25.180 us |    23.553 us |  1468.7500 |  140.6250 |   9,038 KB |
| .NET Framework 4.7.2 | 10 |  16,182.11 us |   320.831 us |   315.099 us |  1843.7500 |  156.2500 |  11,492 KB |
|             .NET 5.0 | 50 | 329,054.39 us | 1,672.983 us | 1,483.056 us | 40000.0000 | 7000.0000 | 249,728 KB |
| .NET Framework 4.7.2 | 50 | 462,394.07 us | 1,273.735 us | 1,129.133 us | 50000.0000 | 2000.0000 | 308,181 KB |
```

File: Tests/FluentAssertions.Specs/Equivalency/CollectionEquivalencySpecs.cs (hunk 1 above)
Comment (dennisdoomen) [reply Tests/FluentAssertions.Specs/Equivalency/CollectionEquivalencySpecs.cs:2835]: Yeah, AppVeyor is a lot slower. Had to increase the threshold to 20 seconds.

**Comment by dennisdoomen** [comment]
All in all, I think it would be nice if we can come up with some approach to fail the build if those benchmarks exceed a certain value.

**Comment by lg2de** [comment]
@jnyrup, you created a benchmark but did not added it to the solution?
Or was THIS benchmark already existing?

**Comment by jnyrup** [comment]
@lg2de I used the example in #1657 for my benchmark, but didn't add it.
Good point about adding it - I'll do that.

**Comment by jnyrup** [comment]
I was just about to add the benchmarks, but seems my former self already did that in #1661.

**Review by mdentremont — COMMENTED** [review]
Good catch and fix!

**Pull request merged**